File: April2022/C++/Top_K_Frequent_Elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> dict;
        for(auto& i:nums) dict[i]++;
        priority_queue<pair<int,int>> q;
        for(auto& i:dict) q.push({i.second,i.first});
        
        vector<int> res;
        while(k--)
        {
            res.push_back(q.top().second);
            q.pop();
        }
        return res;
    }
};

File: April2022/Kotlin/TopKFrequentElements.kt
// Author: Shobhit Behl (LC: shobhitbruh)
fun topKFrequent(nums: IntArray, k: Int): IntArray {
    val map = mutableMapOf<Int, Int>()
    nums.forEach { map[it] = map.getOrDefault(it, 0) + 1 }

    val uniques = mutableListOf<Int>()
    val pairs = map.toList().sortedByDescending { it.second }
    pairs.forEach {
        while (uniques.size < k) {
            uniques.add(it.first)
            break
        }
    }

    return uniques.toIntArray()
}